packages/gatsby/src/internal-plugins/webpack-theme-component-shadowing/index.js

```diff
@@ -50,15 +50,15 @@ module.exports = class GatsbyThemeComponentShadowingResolverPlugin {
           {},
           callback
         )
+      } else {
+        return resolver.doResolve(
```

Comment:
We could change this if into: (just a suggestion)
```
return resolver.doResolve(
  `describedRelative`,
  { ...request, path: builtComponentPath || request.path },
  null,
  {},
  callback
)
```

Reply:
did this